projection/core/src/multi-jvm/scala/com/lightbend/lagom/internal/projection/ProjectionRegistrySpec.scala

```diff
@@ -351,19 +346,20 @@ class ProjectionRegistrySpec
   private def registerProjection(
       projectionName: String,
       tagNames: Set[String],
-      runInRole: Option[String] = None
-  ): TestProbe = {
-    val testProbe = TestProbe()
+  ): Unit =
+    registerProjection(projectionName, tagNames, TestProbe())
 
+  private def registerProjection(
+      projectionName: String,
+      tagNames: Set[String],
+      testProbe: TestProbe,
+  ): Unit = {
     val workerProps = (workerCoordinates: WorkerCoordinates) =>
       FakeProjectionActor.props(
         workerCoordinates.tagName,
         testProbe
       )
-
-    projectionRegistry.registerProjection(projectionName, tagNames, workerProps, runInRole)
-
-    testProbe
+    projectionRegistry.registerProjection(projectionName, tagNames, workerProps)
```

Comment:
That was the issue. 

We create a `TestProber` to pass to `FakeProjectionActor`, then we start the projection and return the probe. 

Outside that method, we were calling `testProbe.ignoreMsg`, but at that point some messages where already delivered to the probe and our assertions later will fail.